### PR TITLE
Update SET to take timeout or options array

### PR DESCRIPTION
**Pull request by gm-ghanover**
Per phpredis documentation:
_Timeout or Options Array_ (optional). If you pass an integer, phpredis will redirect to SETEX, and will try to use Redis >= 2.6.12 extended options if you pass an array with valid values

https://github.com/phpredis/phpredis/#set

### CODE REVIEW

File: tests/units/RedisMock.php [hunk 1]
```diff
@@ -129,6 +129,37 @@ public function testSetGetDelExists()
...
+            ->isEqualTo('OK');
+
+        //set with nx ex
+        $this->assert
+            ->string($redisMock->set('test-set-nx-ex', 'value', ['nx', 'ex' => 1]))
+            ->isEqualTo('OK');
+        $this->assert
+            ->integer($redisMock->set('test-set-nx-ex', 'value', ['nx', 'ex' => 1]))
+            ->isEqualTo(0);
+        sleep(2);
```
Comment (b-viguier) [inline tests/units/RedisMock.php]: Is it for debugging purpose? 😄 
Can you remove it?

File: src/M6Web/Component/RedisMock/RedisMock.php [hunk 2]
```diff
@@ -79,6 +79,22 @@ public function get($key)
...
+            }
+            if (\in_array('xx', $options, true) && !$this->get($key)) {
+                return $this->returnPipedInfo(0);
+            }
+
+            $seconds = null;
+            if (isset($options['ex'])) {
+                $seconds = $options['ex'];
+            } elseif (isset($options['px'])) {
+                $seconds = $options['px'] / 1000;
```
Comment (b-viguier) [inline src/M6Web/Component/RedisMock/RedisMock.php]: Can you add a unit to ease maintenance? Something like `1000 /* ms per seconds*/`?
(In don't know if my example is correct, it's just an example ^^)

File: tests/units/RedisMock.php (hunk 1 above)
Comment (gm-ghanover) [reply tests/units/RedisMock.php]: The test is specifically waiting for an expiration, so you need to sleep for 2 seconds to allow the expiration of 1 second to pass.

Comment (b-viguier) [reply tests/units/RedisMock.php]: OH… my bad, sorry 😅 👍

File: src/M6Web/Component/RedisMock/RedisMock.php (hunk 2 above)
Comment (gm-ghanover) [reply src/M6Web/Component/RedisMock/RedisMock.php]: Well I don't know why the build fails now when all I did was add a comment block...

**Comment by omansour** [comment]
@gm-ghanover possible to rebase the master on this PR ?

**Comment by omansour** [comment]
v4.6.0

**Pull request merged**